### PR TITLE
Fix gcloud retries cmd to rightly capture return code

### PR DESCRIPTION
**Pull request by shyamjvs**
Fixes https://github.com/kubernetes/kubernetes/issues/45489
Seems like the issue was in doing this:
```
local var=$(cmd)
local ret_code=$?  #This actually captures return code of "local" above and not "cmd"
```

cc @wojtek-t @gmarek

### CODE REVIEW

**Comment by shyamjvs** [comment]
This should hopefully fix the problem.

File: test/kubemark/gce/util.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@
 function run-gcloud-compute-with-retries {
   RETRIES="${RETRIES:-3}"
   for attempt in $(seq 1 ${RETRIES}); do
-    local -r gcloud_result=$(gcloud compute "$@" 2>&1)
+    gcloud_result=$(gcloud compute "$@" 2>&1)  # We don't use 'local -r' here as then ret_val seems to always get value 0.
```
Comment (wojtek-t) [inline test/kubemark/gce/util.sh]: Thanks for adding a comment.
Can you please move it to the previous line (to make the line shorter).

**Comment by shyamjvs** [comment]
Sorry, I have no idea why the other commit got included. Let me quickly fix it.

**Comment by wojtek-t** [comment]
Thanks for debugging and fixing!

/lgtm

**Comment by shyamjvs** [comment]
It seems like there's a catch here. Since we have removed the `local -r`, that statement now evaluates to the actual exit code of the gcloud compute command. So if it fails, the script exits with the same code.
Will try to think of a way to solve this.

**Comment by shyamjvs** [comment]
Doing an 'or' of the ret_val assignment right after the gcloud command seems to work (I've tested locally). Let's see what Jenkins has to say.

**Comment by shyamjvs** [comment]
Seems like the add-metadata command wasn't getting the project name right. Fixed it in a follow-up commit.

**Comment by shyamjvs** [comment]
Yay! It works now.
@wojtek-t Could you PTAL finally?

**Pull request merged**